### PR TITLE
[SYCLomatic] Add support for `cuDeviceCanAccessPeer`

### PR DESCRIPTION
**Pull request by the-slow-one**
If the peer-to-peer support extension is enabled, migrate the API. Otherwise add a comment to enable extension. The API mapping is same as `cudaDeviceCanAccessPeer`.

### CODE REVIEW

**Review by TejaX-Alaghari — APPROVED** [review]
LGTM

File: clang/lib/DPCT/APINamesMemory.inc [hunk 1]
```diff
@@ -613,6 +613,23 @@ ASSIGNABLE_FACTORY(FEATURE_REQUEST_FACTORY(
...
 
+CONDITIONAL_FACTORY_ENTRY(
+    UsePeerAccess(),
+    ASSIGNABLE_FACTORY(ASSIGN_FACTORY_ENTRY(
+        "cuDeviceCanAccessPeer", DEREF(ARG_WC(0)),
+        MEMBER_CALL(CALL(MapNames::getDpctNamespace() +
+                             "dev_mgr::instance().get_device",
+                         ARG_WC(1)),
+                    false, "ext_oneapi_can_access_peer",
+                    CALL(MapNames::getDpctNamespace() +
```
Comment (AndyCHHuang) [inline clang/lib/DPCT/APINamesMemory.inc]: this should be another member call

Comment (the-slow-one) [reply clang/lib/DPCT/APINamesMemory.inc]: We only want to get the second device and pass it to `ext_oneapi_can_access_peer`. And we don't want to access any members from the device returned from `get_device()`.

Therefore, it's just a call.

Comment (AndyCHHuang) [reply clang/lib/DPCT/APINamesMemory.inc]: "dpct::dev_mgr::instance().get_device(device).ext_oneapi_can_access_peer(dpct::dev_mgr::instance().get_device(peerDevice));"

The first dpct::dev_mgr::instance().get_device(device) is a member call.
ext_oneapi_can_access_peer() is another membercall.
The arg "dpct::dev_mgr::instance().get_device(peerDevice)" is also a member call.

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/APINamesMemory.inc (hunk 1 above)
Comment (the-slow-one) [reply clang/lib/DPCT/APINamesMemory.inc]: Replaced CALL with MEMBER_CALL template.

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Pull request merged**